modules/nextflow/src/main/groovy/nextflow/script/ScriptMeta.groovy

```diff
@@ -202,6 +202,16 @@ class ScriptMeta {
         return this
     }
 
+    ScriptMeta addFunctionDefinition(FunctionDef component) {
```

Comment:
I'm wondering why this check is only needed for function imports. What if a user include process `foo` and then a function `foo` from another module? 

Maybe is needed a general check here? 

https://github.com/nextflow-io/nextflow/blob/2982-allow-the-import-for-custom-function-from-plugin-extensions/modules/nextflow/src/main/groovy/nextflow/script/ScriptMeta.groovy#L192-L194

Reply:
True, will create a test to cover this situation